pkg/controller/service/servicecontroller.go

```diff
@@ -339,14 +339,23 @@ func (s *ServiceController) createLoadBalancerIfNeeded(namespacedName types.Name
 
 		// TODO: We could do a dry-run here if wanted to avoid the spurious cloud-calls & events when we restart
 
-		// The load balancer doesn't exist yet, so create it.
-		s.eventRecorder.Event(service, api.EventTypeNormal, "CreatingLoadBalancer", "Creating load balancer")
-
-		err := s.createLoadBalancer(service, namespacedName)
+		_, exists, err := s.balancer.GetLoadBalancer(s.loadBalancerName(service), s.zone.Region)
```

Comment:
I'm confused by your question. Endpoints controller has no knowledge about openstack at all. AFAIK this is the code that manages hosts: https://github.com/kubernetes/kubernetes/blob/master/pkg/cloudprovider/providers/openstack/openstack.go#L727, if you can confirm that it detects that the right hosts are in the pool even when the pool name is the same, that would be great.